### ISSUE TITLE
calling python from go (planetmint modules into go)

### ISSUE THREAD

**Issue by eckelj** [planetmint/planetmint#343]

**Comment by eckelj** [comment]
## Calling python from golang (in memory python)

Nearly all tutorials refer to the following example: [https://github.com/ardanlabs/python-go/tree/master/py-in-mem](https://github.com/ardanlabs/python-go/tree/master/py-in-mem)

The example utilizes the Python-C API from golang.

A nice Python-C wrapper exists at

[https://github.com/go-python/cpy3](https://github.com/go-python/cpy3)

but officially supports only python 3.7. Nevertheless, I suggest utilizing it and for a PoC.

**Comment by eckelj** [comment]
https://www.notion.so/rddlnetwork/Series-of-Ignite-workshops-efdf56043e434cd983a78e95beeef82a?pvs=4#3b17042a144f4494938820df6622c674

**Comment by eckelj** [comment]
https://www.notion.so/rddlnetwork/Calling-Python-from-go-2c6792d13d71411399f6d22927047259 shows how to use python code: with the adjusted cpy3 version.

**Comment by jmastr** [comment]
you are probably further then this already, but for reference:
```
package main

// #cgo pkg-config: python3-embed
// #include <Python.h>
import "C"

import (
        "unsafe"
)

func main() {
        pycodeGo := `
import sys
for path in sys.path:
  print(path)
`

        defer C.Py_Finalize()
        C.Py_Initialize()
        pycodeC := C.CString(pycodeGo)
        defer C.free(unsafe.Pointer(pycodeC))
        C.PyRun_SimpleString(pycodeC)
}
```

**Comment by jmastr** [comment]
This uses pkg-config to find the current libpython3, e.g. on my machine:
```
> pkg-config --cflags --libs -- python3-embed
-I/nix/store/5axq6aw8j3vcs2m7gi440cwpcckl7ql9-python3-3.10.9/include/python3.10 -L/nix/store/5axq6aw8j3vcs2m7gi440cwpcckl7ql9-python3-3.10.9/lib -lpython3.10
```

So it uses my `python3.10` install.

**Comment by eckelj** [comment]
yes! that's what I also have! but with the complete python C APi wrapper from cpy3 adjusted to python 3.9 for the time being.

**Comment by jmastr** [comment]
Personally I like to use the features directly if possible without a wrapper around it. But that's just me I guess.
